hazelcast/src/main/java/com/hazelcast/config/alto/AltoSocketConfig.java

```diff
@@ -4,55 +4,48 @@
 
 import javax.annotation.Nonnull;
 import java.util.Objects;
-import java.util.regex.Pattern;
 
 public class AltoSocketConfig {
-    private static final String DEFAULT_PORT_RANGE = "11000-21000";
-    private static final int DEFAULT_RECEIVE_BUFFER_SIZE = 128 * 1024;
-    private static final int DEFAULT_SEND_BUFFER_SIZE = 128 * 1024;
-
-    private String portRange = DEFAULT_PORT_RANGE;
-    private int receiveBufferSize = DEFAULT_RECEIVE_BUFFER_SIZE;
-    private int sendBufferSize = DEFAULT_SEND_BUFFER_SIZE;
+    private String portRange = "11000-21000";
+    private int receiveBufferSizeKb = 128;
+    private int sendBufferSizeKb = 128;
 
     public String getPortRange() {
         return portRange;
     }
 
     public AltoSocketConfig setPortRange(@Nonnull String portRange) {
-        if (!Bounds.PORT_RANGE_PATTERN.matcher(portRange).matches()) {
-            throw new InvalidConfigurationException("Invalid port definition");
+        if (!portRange.matches("\\d{1,5}-\\d{1,5}")) {
+            throw new InvalidConfigurationException("Invalid port range");
         }
 
         this.portRange = portRange;
         return this;
     }
 
-    public int getReceiveBufferSize() {
-        return receiveBufferSize;
+    public int getReceiveBufferSizeKb() {
+        return receiveBufferSizeKb;
     }
 
-    public AltoSocketConfig setReceiveBufferSize(int receiveBufferSize) {
-        if (receiveBufferSize < Bounds.MIN_BUFFER_SIZE || receiveBufferSize > Bounds.MAX_BUFFER_SIZE) {
-            throw new InvalidConfigurationException("Buffer size should be between "
-                    + Bounds.MIN_BUFFER_SIZE + " and " + Bounds.MAX_BUFFER_SIZE);
+    public AltoSocketConfig setReceiveBufferSizeKb(int receiveBufferSizeKb) {
+        if (receiveBufferSizeKb < 32 || receiveBufferSizeKb > 1_048_576) {
+            throw new InvalidConfigurationException("Buffer size should be between " + 32 + " and " + 1_048_576);
         }
 
-        this.receiveBufferSize = receiveBufferSize;
+        this.receiveBufferSizeKb = receiveBufferSizeKb;
         return this;
     }
 
-    public int getSendBufferSize() {
-        return sendBufferSize;
+    public int getSendBufferSizeKb() {
+        return sendBufferSizeKb;
     }
 
-    public AltoSocketConfig setSendBufferSize(int sendBufferSize) {
-        if (sendBufferSize < Bounds.MIN_BUFFER_SIZE || sendBufferSize > Bounds.MAX_BUFFER_SIZE) {
-            throw new InvalidConfigurationException("Buffer size should be between "
-                    + Bounds.MIN_BUFFER_SIZE + " and " + Bounds.MAX_BUFFER_SIZE);
+    public AltoSocketConfig setSendBufferSizeKb(int sendBufferSizeKb) {
+        if (sendBufferSizeKb < 32 || sendBufferSizeKb > 1_048_576) {
```

Comment:
Only check for positive. Don't put weird constraints on things where not needed.

Reply:
If you have a network with a latency (Long Fat Network), you need to have huge network buffers to be able to utilize the network bandwidth. The size of the buffers is calculated using the BDP: Bandwidth Delay Product. Having buffers of max 1 MB will lead to terrible performance on such environments.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/16aaf911c270da78592443f9d81c145858b9459e